### PR TITLE
Implement /detect/generated

### PR DESCRIPTION
**Pull request by mdevino**
This addresses #151.

Scenarios tested:

* Sample request -> works just fine (base success scenario).
* Missing `prompt` -> returns 422
* Missing `generated_text` -> returns 422
* Missing `detectors` -> returns 422
* Two detectors -> returns detections from both detectors (though the order of detections may vary)
* Threshold filtering -> only detections above the threshold are returned
* Detector returning 404 -> returns 404
* Detector returning 503 -> returns 503
* Detector returning 500 (further details are shown on logs) -> returns 500
* Detector returning 422 -> returns 422

### CODE REVIEW

File: src/models.rs [hunk 1]
```diff
@@ -985,6 +985,47 @@ pub struct ContextDocsResult {
...
+}
+
+/// The response format of the /api/v2/text/detection/generated endpoint
+#[derive(Default, Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct DetectionOnGenerationResult {
+    /// Detection results
+    #[serde(rename = "detections")]
+    pub detections: Vec<DetectionResult>,
+}
+
```
Comment (mdevino) [inline src/models.rs:1028]: This structure has the same field as `ContextDocsResult`, but I coudn't think of a good name to reuse the already existing one. Any suggestions?

File: src/orchestrator/unary.rs [hunk 2]
```diff
@@ -313,6 +313,65 @@ impl Orchestrator {
             }
         }
     }
+
+    /// Handles detections on generated textual context
```
Comment (gkumbhat) [inline src/orchestrator/unary.rs:317]:
```suggestion
    /// Handles detections on generated text (without performing generation)
```

**Review by gkumbhat — COMMENTED** [review]
Small comment. Otherwise looks good. Thanks Mateus

File: src/models.rs (hunk 1 above)
Comment (mdevino) [reply src/models.rs:1028]: Sure.

File: docs/api/orchestrator_openapi_0_1_0.yaml [hunk 3]
```diff
@@ -400,6 +434,44 @@ components:
...
+        detectors:
+          type: object
+          title: Detectors
+          default: {}
+          example: 
+            generated-detection-v1-model-en: {}
+      type: object
+      required: ["generated_text", "prompt", "detectors"]
+      title: Generated-Text Detection Request
+    GeneratedTextDetectionResponse:
```
[the remark this answers is not in the document]
Comment (gkumbhat) [reply docs/api/orchestrator_openapi_0_1_0.yaml:454]: Yeah, the reason for duplicating it here is with speculation that these differnt endpoint would end up having different parameter and response elements in future. We started with similar and simple ones, but things may expand in this direction..

File: src/server.rs [hunk 4]
```diff
@@ -163,6 +164,10 @@ pub async fn run(
             &format!("{}/detection/context", TEXT_API_PREFIX),
             post(detect_context_documents),
         )
+        .route(
+            &format!("{}/detection/generated", TEXT_API_PREFIX),
```
[the remark this answers is not in the document]
Comment (mdevino) [reply src/server.rs:168]: Sorry, the notification for this request didn't ring. I see that Gaurav already fixed it, thought. Thanks!

**Pull request merged**